### ISSUE TITLE
Incorrect TextPath display under interactive mode

### ISSUE THREAD

**Issue by qinggeng** [matplotlib/matplotlib#6049]
matplotlib version 1.5.1 installed from pip
osx 10.9.5
python version: 2.7.11 installed from source code

source code:
```python
from pylab import *
from matplotlib.text import TextPath
from matplotlib.patches import PathPatch
t = TextPath((0, 20), u'ABCDEFGHIJK')
tp = PathPatch(t, color='k', fill = False)
plt.gca().set_xlim(-100, 100)
plt.gca().set_ylim(0, 50)
plt.gca().add_artist(tp)
plt.show()
````
result screenshot:
![mac_display_bug](https://cloud.githubusercontent.com/assets/1065213/13240597/d1ff8118-da1e-11e5-8fe0-2f2d35f97265.png)

However, the saved image is correct(saved from save button):
![mpl_hollo_textpathbug](https://cloud.githubusercontent.com/assets/1065213/13240749/14f54402-da20-11e5-8f80-88992c322e38.png)

**Comment by WeatherGod** [comment]
Wow, I never knew this feature existed!

I just tried it against master and v1.5.1, and couldn't reproduce it on my
linux box. Perhaps it is a macosx backend bug? Could you try a different
backend like Qt4Agg?

**Comment by mdboom** [comment]
I can confirm @WeatherGod's hunch that this is a macosx-backend specific bug.  We have medium-term plans to replace the macosx backend with one that does its rendering with Agg to get around these kinds of things.

**Comment by mdboom** [comment]
Fixed by #6178
